internal/resource/resourcetest/acls.go

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package resourcetest
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"google.golang.org/protobuf/reflect/protoreflect"
+
+	"github.com/hashicorp/consul/acl"
+	"github.com/hashicorp/consul/agent/structs"
+	"github.com/hashicorp/consul/internal/resource"
+	"github.com/hashicorp/consul/proto-public/pbresource"
+)
+
+const (
+	DENY    = "deny"
+	ALLOW   = "allow"
+	DEFAULT = "default"
+)
+
+var checkF = func(t *testing.T, expect string, got error) {
+	switch expect {
+	case ALLOW:
+		if acl.IsErrPermissionDenied(got) {
+			t.Fatal("should be allowed")
+		}
+	case DENY:
+		if !acl.IsErrPermissionDenied(got) {
+			t.Fatal("should be denied")
+		}
+	case DEFAULT:
+		require.Nil(t, got, "expected fallthrough decision")
+	default:
+		t.Fatalf("unexpected expectation: %q", expect)
+	}
+}
+
+type ACLTestCase struct {
+	Rules   string
+	Data    protoreflect.ProtoMessage
+	Owner   *pbresource.ID
+	Typ     *pbresource.Type
+	ReadOK  string
+	WriteOK string
+	ListOK  string
+}
+
+func RunACLTestCase(t *testing.T, tc ACLTestCase, registry resource.Registry) {
+	reg, ok := registry.Resolve(tc.Typ)
+	require.True(t, ok)
+
+	res := Resource(tc.Typ, "test").
+		WithTenancy(resource.DefaultNamespacedTenancy()).
```

Comment:
I believe we can make this helper more generic by getting this from the registry.